hibernate-core/src/main/java/org/hibernate/internal/util/config/ConfigurationHelper.java

```diff
@@ -220,8 +220,8 @@ public static long getLong(String name, Map values, int defaultValue) {
 		if ( value == null ) {
 			return defaultValue;
 		}
-		if ( Long.class.isInstance( value ) ) {
-			return (Long) value;
+		if ( Number.class.isInstance( value ) ) {
+			return ( (Number) value ).longValue();
```

Comment:
This is a generic code change in configuration. unsure whether there is some unexpected implication. @gavinking maybe you would be interested to take a look.

Reply:
Additionally, I will suggest to change type of defaultValue parameter from int to long.